src/sys/shell/mod.rs

```diff
@@ -16,12 +16,32 @@ cfg_tcp! {
 }
 
 cfg_udp! {
-    mod udp;
-    pub(crate) use self::udp::UdpSocket;
+    pub mod udp;
 }
 
 #[cfg(unix)]
 cfg_uds! {
     mod uds;
     pub(crate) use self::uds::{UnixDatagram, UnixListener, UnixStream};
 }
+
+cfg_net! {
```

Comment:
`IoSourceState` and it's methods are only used by UDP, so similarly this should be `cfg_udp!`